### PR TITLE
Interpolation search

### PR DESCRIPTION
**Pull request by Akshay2903**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [x] **If creating a new file :**
  - [x] added links to it in the README files ?
  - [x] included tests with it ?
  - [ ] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

**Comment by keon** [comment]
thanks! could you add some tests?

**Comment by Akshay2903** [comment]
I add the test case

File: algorithms/search/interpolation_search.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# interpolation search work for sorted list
+
+def interpolation_search(numbersList,value):
+    low = 0
+    high = len(numbersList) - 1
+    while numbersList[low] <= value and numbersList[high] >= value:
```
Comment (keon) [inline algorithms/search/interpolation_search.py]: use camel case -> numbers_list

File: algorithms/search/interpolation_search.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            return mid
+    if numbersList[low] == value:
+        return low
+    return None
+
+def test(x):
+    arr = [10, 12, 13, 16, 18, 19, 20, 21, 22, 23, 24, 33, 35, 42, 47]
+    index = interpolation_search(arr, x)
+    if index != -1:
+        print ("The element", x, "is at the index", index)
```
Comment (keon) [inline algorithms/search/interpolation_search.py]: do not print. instead use assert as other tests

**Comment by keon** [comment]
you need to put the test in a separate file.
take these tests as examples:
you can put it inside `search_test.py`
https://github.com/keon/algorithms/tree/master/tests

**Comment by ericklarac** [comment]
Closing PR because this algorithm has been recently added and merged.
